quota/common/lib/memory_buffer.js

```diff
@@ -215,13 +216,22 @@ Bucket.prototype.flushBucket = function(cb) {
 
     var sameTimeBucket = (self.expires === localExpires) &&                        // same local time bucket?
                          (!remoteExpires || remoteExpires === self.remoteExpires); // same remote time bucket?
+
+    if ( reply.expiryTimestamp ) { // filter out back dated remote expiry, these are invalid responses from edge
+      sameTimeBucket = sameTimeBucket && ( !self.remoteExpiryTimestamp || reply.expiryTimestamp >= self.remoteExpiryTimestamp );
```

Comment:
Change comment - the responses can be from any remote, not just edge
verify the sameTimeBucket condition

Reply:
Comment change done.

Reply:
reply.expiryTimestamp >= self.remoteExpiryTimestamp condition is working